recursion.rb

```diff
@@ -8,15 +8,60 @@ def factorial(n)
   end
 end
 
+# ### Write #2
+# `reverse(s)`
+# Write a method `reverse` that accepts a string as a parameter
+# and then returns the reverse of the string.
+#
+# - e.g. reverse("hello") = "olleh"
+
 def reverse(s)
+  if s.length == 0
+    return ""
+  else
+    return s[-1] + reverse(s[0..-2])
+  end
 end
 
+# ### Write #3
+# Write a method `bunny` that accepts an integer parameter n. N represents a
+# number of bunnies and each bunny has two big floppy ears. We want to compute the total number of ears across all the bunnies recursively (without loops or multiplication).
+#
+# - e.g. bunny(0) = 0
+# - e.g. bunny(1) = 2
+# - e.g. bunny(10) = 20
+
 def bunny(n)
+  if n == 0
+    return 0
+  else
+    return 2 + bunny(n-1)
+  end
 end
 
+
+
+# ### Write #4
+# `nested(s)`
+# Write a method `nested` that accepts a string of only parenthesis
+# and then returns if those parenthesis are properly nested. You may
+# assume that no non-paren characters will be passed to this method.
+#
+# - e.g. nested("((()))") = true
+# - e.g. nested("())") = false
+
 def nested(s)
+  if s.length == 0
+    return true
+  elsif s[0] == "(" && s[-1] == ")"
+    return nested(s[1..-2])
+  else
+    return false
+  end
 end
```

Comment:
I think you might want to add one more base case here that checks for `s.length == 1`. 

irb tells me that something like `"hi"[3]` is `nil` and that `nil && true` is `nil` ... anyway, Ruby may handle this case fine with fancy ruby stuff, but other languages would warrant another check, so I'd add it in.